### DISCUSSION TITLE
Face Similarity

### DISCUSSION THREAD

**Discussion by rs2-davebanguilan** [vladmandic/face-api#123] [Q&A]
I'm trying to create a backend using this package that can detect if a picture on a user's ID and a user's selfie is the same. My problem for now is what will be my threshold to determine if two faces is the same, different, or unsure (need to check manually). Here's my [service]( https://gist.github.com/rs2-davebanguilan/925d09af72ba05a05d47883348126f77) on my nodejs project You can check the `calcSimilarityPercentage` function. Also, am I using the functions right? the backend could only receive 2 image, 1 for the selfie and 1 for the picture of an ID with user's photo on it (Like a driver's license).

**Comment by vladmandic** [comment]
your `getDescriptors` method looks good, but if you only want to compare two faces, you can skip entire `faceMatcher` class as its pretty much a helper class that wraps a lot of things you don't need - you can simply calculate feature vector distance directly

```
const desc1 = await getDescriptors(file1);
const desc2 = await getDescriptors(file2);
const distance = faceapi.euclideanDistance(desc1[0], desc2[0]); // only compare first found face in each image
```
